$MirOS: ports/games/frozen-bubble/patches/patch-c_stuff_Makefile_PL,v 1.2 2006/11/04 03:09:20 tg Exp $
$NetBSD: patch-aa,v 1.4 2003/04/18 08:07:51 salo Exp $
--- c_stuff/Makefile.PL.orig	Thu Oct 19 20:46:12 2006
+++ c_stuff/Makefile.PL	Sat Nov  4 04:09:32 2006
@@ -16,7 +16,7 @@ $main;
 }
 EOTEST
 	close(CTEST);
-	$r = system("$Config{cc} -o $fname $fname.c $cflags $libs 2>/dev/null 1>/dev/null");
+	$r = system("$Config{cc} -o $fname $fname.c $cflags $libs >>res 2>&1");
 	warn("RUNNING: $Config{cc} -o $fname $fname.c $cflags $libs\n") if $ENV{VERBOSE};
 	unlink($fname, "$fname.c");
 	return $r == 0;
@@ -34,8 +34,8 @@ system("sdl-config --cflags 2>/dev/null 
 ccompile('#include <SDL.h>
           #include <SDL_mixer.h>',
 	 '',
-	 chomp_(`sdl-config --cflags`),
-	 chomp_(`sdl-config --libs`) . ' -lSDL_mixer')
+	 chomp_($ENV{CFLAGS} . ' ' . $ENV{CPPFLAGS} . ' ' . `sdl-config --cflags`),
+	 chomp_($ENV{LDFLAGS} . ' ' . `sdl-config --libs`) . ' -lSDL_mixer -lvorbisfile -lvorbis -logg')
   or
   die_ 'SDL_mixer is needed (and, version >= 1.2.2)';
 
@@ -43,8 +43,8 @@ ccompile('#include <SDL.h>
 ccompile('#include <SDL.h>
           #include <SDL_mixer.h>',
 	 'Mix_FadeInMusicPos(NULL, 0, 0, 0);',
-	 chomp_(`sdl-config --cflags`),
-	 chomp_(`sdl-config --libs`) . ' -lSDL_mixer')
+	 chomp_($ENV{CFLAGS} . ' ' . `sdl-config --cflags`),
+	 chomp_($ENV{LDFLAGS} . ' ' . `sdl-config --libs`) . ' -lSDL_mixer -lvorbisfile -lvorbis -logg')
   or
   die_ 'SDL_mixer >= 1.2.2 is needed (impossible to create an executable with function Mix_FadeInMusicPos)';
 
@@ -77,6 +77,6 @@ WriteMakefile(
     'VERSION_FROM' => 'fb_c_stuff.pm', # finds VERSION
     'OBJECT'       => 'fb_c_stuff.o',
     'INC'          => chomp_(`sdl-config --cflags`) . ' ' . chomp_(`pkg-config SDL_Pango --cflags`) . ' -I.',
-    'OPTIMIZE'     => '-O2 -Wall',
+    'OPTIMIZE'     => chomp_('-O2 -Wall' . $ENV{CFLAGS}),
     'MAKEFILE'     => 'Makefile_c',
 );
